poc_mobile_verifier/src/cli/generate.rs

```diff
@@ -18,37 +16,61 @@ pub struct Cmd {
     #[clap(long)]
     before: NaiveDateTime,
     #[clap(long)]
-    input_bucket: String,
+    periods: i32,
 }
 
 impl Cmd {
     pub async fn run(self) -> Result {
         let Self {
             after,
             before,
-            input_bucket,
+            periods,
         } = self;
 
-        tracing::info!(
-            "Verifying shares from bucket {input_bucket} within the following time range: {after} to {before}"
-        );
+        let after = DateTime::from_utc(after, Utc);
+        let before = DateTime::from_utc(before, Utc);
+
+        tracing::info!("Verifying shares from the following time range: {after} to {before}");
 
-        let input_store = FileStore::new(None, "us-west-2", input_bucket).await?;
+        let file_store = FileStore::from_env().await?;
 
-        let follower_service = follower::Client::new(
+        let follower = follower::Client::new(
             Endpoint::from(env_var("FOLLOWER_URI", Uri::from_static(DEFAULT_URI))?)
                 .connect_timeout(CONNECT_TIMEOUT)
                 .timeout(RPC_TIMEOUT)
                 .connect_lazy(),
         );
 
-        let rewards = SubnetworkRewards::from_period(
-            &input_store,
-            follower_service,
-            DateTime::from_utc(after, Utc),
-            DateTime::from_utc(before, Utc),
-        )
-        .await?;
+        let db_connection_str = dotenv::var("DATABASE_URL")?;
+        let pool = PgPoolOptions::new()
+            .max_connections(10)
+            .connect(&db_connection_str)
+            .await?;
+        sqlx::migrate!().run(&pool).await?;
+
+        let mut verifier = Verifier::new(&pool, file_store, follower).await?;
+
+        let mut transaction = pool.begin().await?;
+        let reward_period = before - after;
+        let verification_period = reward_period / periods;
+        let mut now = after;
+
+        for period in 0..periods {
+            let verify_epoch = verifier.get_verify_epoch(now);
+            tracing::info!("Verifying epoch {period} of {periods}...");
+            let _ = verifier
+                .verify_epoch(&mut transaction, verify_epoch)
+                .await?;
+            now += verification_period;
+        }
```

Comment:
would this for loop with an external mutable index be better expressed as an iter-fold over the periods?